src/server/db_slice.h

```diff
@@ -62,6 +62,42 @@ class DbSlice {
   void operator=(const DbSlice&) = delete;
 
  public:
+  class AutoPostUpdate {
+   public:
+    AutoPostUpdate();
+    AutoPostUpdate(AutoPostUpdate&& o);
+    AutoPostUpdate& operator=(AutoPostUpdate&& o);
+    ~AutoPostUpdate();
+
+    void Run();
+    void Cancel();
+
+   private:
+    enum class DestructorAction {
+      kDoNothing,
+      kRun,
+    };
+
+    // Wrap members in a struct to auto generate operator=
```

Comment:
The reason you actually need this field is to clear it in the move constructor instead of dropping in an undefined state 🤔 Not handy, but I don't know any other solution instead of having a move-clearable flag

Reply:
What do you mean by a move-clearable flag?
(I don't see any issue with this implementation though..)

Reply:
Just an interesting thought I had... So this example

```c++
struct PostUpdater {
  PostUpdater& operator=(PostUpdater&& o) = default;

  ~PostUpdater() { cout << a << " and " << uint64_t(b.get()) << endl; }

  int a;
  unique_ptr<int> b;
};

int main() {
  PostUpdater p1(1, make_unique<int>(11));
  PostUpdater p2(2, nullptr);

  p2 = std::move(p1);
}
```

will print

```
1 and 94585575722672
1 and 0
```

because moving an int is just copying it.... A move-clearable flag would behave like unique_ptr<bool> (but without the heap allocation). This way to don't need the nested struct, but can rely on default move constructor / operator to store "default" values in the moved-from object that can then be checked inside the destructor

Reply:
> Gotcha. Well, it's simpler currently :)

Not arguing, just if we or absl had this type, it'd be one line